streams/examples/main/java/org/apache/kafka/streams/examples/WordCount/WordCountJob.java

```diff
@@ -0,0 +1,106 @@
+
+import java.util.Arrays;
+import java.util.Properties;
+
+public class WordCountJob {
+
+    public static void main(String[] args) throws Exception {
+        Properties props = new Properties();
+        props.put(StreamingConfig.JOB_ID_CONFIG, "streams-wordcount");
+        props.put(StreamingConfig.STATE_DIR_CONFIG, "/tmp/streams");
+        props.put(StreamingConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+        props.put(StreamingConfig.ZOOKEEPER_CONNECT_CONFIG, "localhost:2181");
+        props.put(StreamingConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+        props.put(StreamingConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+        props.put(StreamingConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        props.put(StreamingConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+
+        // can specify underlying client configs if necessary
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        StreamingConfig config = new StreamingConfig(props);
+
+        KStreamBuilder builder = new KStreamBuilder();
+
+        final Serializer<String> stringSerializer = new StringSerializer();
+        final Deserializer<String> stringDeserializer = new StringDeserializer();
+        final Serializer<Long> longSerializer = new LongSerializer();
+        final Deserializer<Long> longDeserializer = new LongDeserializer();
+        final Serializer<JsonNode> JsonSerializer = new JsonSerializer();
+
+        KStream<String, String> source = builder.stream("streams-file-input");
+
+        KStream<String, JsonNode> counts = source
+                .flatMapValues(new ValueMapper<String, Iterable<String>>() {
+                    @Override
+                    public Iterable<String> apply(String value) {
+                        return Arrays.asList(value.toLowerCase().split(" "));
+                    }
+                }).map(new KeyValueMapper<String, String, KeyValue<String, String>>() {
+                    @Override
+                    public KeyValue<String, String> apply(String key, String value) {
+                        return new KeyValue<String, String>(value, value);
+                    }
+                })
+                .aggregateByKey(new Count<>(), UnlimitedWindows.of("Counts").startOn(0L),
+                        stringSerializer, longSerializer,
+                        stringDeserializer, longDeserializer)
+                .toStream()
+                .map(new KeyValueMapper<Windowed<String>, Long, KeyValue<String, JsonNode>>() {
+                    @Override
+                    public KeyValue<String, JsonNode> apply(Windowed<String> key, Long value) {
+                        ObjectNode jNode = JsonNodeFactory.instance.objectNode();
+
+                        jNode.put("word", key.value())
+                                .put("count", value);
+
+                        return new KeyValue<String, JsonNode>("", jNode);
```

Comment:
Why would this be an empty string? I would have expected it to be either null (e.g., we don't care about preserving the key at all) or `key.value()` (e.g., we want to use a compacted topic and only preserve the latest count since it is the most recent count over the infinite window).